src/app/api/auth/[...nextauth]/route.ts

```diff
@@ -2,9 +2,27 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import { NextApiRequest, NextApiResponse } from "next";
+import { NextResponse } from "next/server";
 import NextAuth from "next-auth";
 import { authOptions } from "../../utils/auth";
 
-const handler = NextAuth(authOptions);
+// There is currently no support for handling OAuth provider callback errors:
+// https://github.com/nextauthjs/next-auth/discussions/8209
+const handler = async (
+  req: Request & NextApiRequest,
+  res: Request & NextApiResponse,
+) => {
+  if (
+    req.method === "GET" &&
+    req.url?.startsWith(
+      `${process.env.SERVER_URL}/api/auth/callback/fxa?error=`,
+    )
+  ) {
+    return NextResponse.redirect(`${process.env.SERVER_URL}/user/dashboard`);
```

Comment:
@flozia Shouldn't this be going to `/` rather than `/user/dashboard/`? Now, if I'm not signed in, I get redirected to FxA, then back to Monitor, then to the dashboard, where I see a flash of an error page, before being redirected back to FxA, but this time not a `prompt=none` flow, so I'm still asked to sign in. (If that's what we want, we could just not do a `prompt=none` flow, right?)

Reply:
@Vinnl Redirecting to / might be safer in this case. The idea behind redirecting to /user/dashboard was to initiate authentication as a fallback, but there might be too many redirects — I’ve opened PR #5192.